frontend/src/app/auth.service.ts

```diff
@@ -19,12 +19,12 @@ export class AuthService {
   private logoutUrl = `${environment.apiRoot}/logout/`;
   private resetPasswordUrl = `${environment.apiRoot}/password-reset/`;
 
-  private changeUserEvent = new BehaviorSubject<User | null>(null);
+  private changeUserEvent = new ReplaySubject<User>(1);
   private loginEvent = new Subject<SuccessOrFailureAction>();
   private resetPasswordEvent = new Subject<SuccessOrFailureAction>();
   private confirmResetPasswordEvent = new Subject<SuccessOrFailureAction>();
 
-  public user$: Observable<User | null> = this.changeUserEvent.asObservable();
+  public user$: Observable<User> = this.changeUserEvent.asObservable();
```

Comment:
Same here with the type: from reading the code it seems that it is still <User | null > type